cardano-api/src/Cardano/Api/IPC/Monad.hs

```diff
@@ -45,14 +46,14 @@ import           Cardano.Api.Modes
 -- In order to make pipelining still possible we can explore the use of Selective Functors
 -- which would allow us to straddle both worlds.
 newtype LocalStateQueryExpr block point query r m a = LocalStateQueryExpr
-  { runLocalStateQueryExpr :: ContT (Net.Query.ClientStAcquired block point query m r) m a
-  } deriving (Functor, Applicative, Monad, MonadIO)
+  { runLocalStateQueryExpr :: ReaderT NodeToClientVersion (ContT (Net.Query.ClientStAcquired block point query m r) m) a
+  } deriving (Functor, Applicative, Monad, MonadReader NodeToClientVersion, MonadIO)
 
 -- | Execute a local state query expression.
 executeLocalStateQueryExpr
   :: LocalNodeConnectInfo mode
   -> Maybe ChainPoint
-  -> (NodeToClientVersion -> LocalStateQueryExpr (BlockInMode mode) ChainPoint (QueryInMode mode) () IO a)
```

Comment:
The use of `ReaderT` allows this argument to simplify.  This means that a query can be run like this:

```
executeLocalStateQueryExpr ... $ do
    ...
    expr
    ...
```

instead of like this:

```
executeLocalStateQueryExpr ... $ \ntcVersion -> do
    ...
    expr
    ...
```

We are transitioning from the later code because we want the `ntcVersion` to be checked automatically rather than manually.  The `ntcVersion` argument was passed in via a lambda before because the monadic query expression previously check the version manually which is error prone.